Add FLASH_SIMULATOR to board xenvm and xenvm-qemu

dsemenets opened the pull request.

oleksiimoisieiev commented on boards/arm64/xenvm/xenvm_defconfig:
```diff
@@ -33,9 +34,19 @@ CONFIG_MAIN_STACK_SIZE=32768
-CONFIG_CMSIS_V2_THREAD_MAX_STACK_SIZE=32768
-CONFIG_CMSIS_V2_THREAD_DYNAMIC_STACK_SIZE=32768
-CONFIG_IPM_CONSOLE_STACK_SIZE=32768
+#CONFIG_CMSIS_THREAD_MAX_STACK_SIZE=32768
```
please - no commented code

oleksiimoisieiev commented on soc/arm64/xenvm/soc.h:
```diff
@@ -0,0 +1,4 @@
+#ifndef _SOC_H
```
Still don't understand why do we need this header

dsemenets replied:
We need this header because we need FLASH_MAP which need this header

oleksiimoisieiev replied:
I've seached for soc.h in soc/arm64
The result is the following:
(ranger) ➜  arm64 git:(xrun) ✗ fnd soc.h
./nxp_imx/mimx8m/pinctrl_soc.h
./nxp_imx/mimx9/pinctrl_soc.h
./bcm_vk/viper/soc.h

where directory contents are: 
arm  bcm_vk  CMakeLists.txt  intel_socfpga  Kconfig  nxp_imx  nxp_layerscape  qemu_cortex_a53  qemu_virt_arm64  rockchip  xenvm

As you can see - not all socs have this header. Does it mean that qemu_cotrex_a53 for example, does not support FLASH?
There should be another way.

lorc commented on boards/arm64/xenvm/xenvm_defconfig:
```diff
@@ -33,9 +33,15 @@ CONFIG_MAIN_STACK_SIZE=32768
-CONFIG_CMSIS_THREAD_MAX_STACK_SIZE=32768
-CONFIG_CMSIS_V2_THREAD_MAX_STACK_SIZE=32768
-CONFIG_CMSIS_V2_THREAD_DYNAMIC_STACK_SIZE=32768
-CONFIG_IPM_CONSOLE_STACK_SIZE=32768
```
Looks like spurious change. If you really need to remove all those lines - put them into separate commit.

lorc commented on boards/arm64/xenvm/xenvm_defconfig:
```diff
@@ -13,7 +13,7 @@ CONFIG_CONSOLE=y
 
 # Enable PV hypervisor console
-CONFIG_UART_CONSOLEIO_XEN_HVC=y
+CONFIG_UART_XEN_HVC_CONSOLEIO=y
```
I know about that fix and it should go into another commit. It does nothing with virtual flash

lorc commented on boards/arm64/xenvm/xenvm-qemu_defconfig:
```diff
@@ -24,8 +24,15 @@ CONFIG_LOG=y
 CONFIG_LOG_MODE_MINIMAL=n
 
 CONFIG_TEST=y
-#CONFIG_PRINTK_SYNC=y
```
Should be in a separate commit, IMO

lorc reviewed (commented):
Please don't add anything unrelated to the commit. Remember that commits should be atomic. Also, you really need to write proper commit description.

lorc commented:
Also, there are some auto-checks failed. You need to address this. Except Manifest check.

firscity commented on boards/arm64/xenvm/xenvm-qemu.dts:
```diff
@@ -82,4 +83,45 @@
 		compatible = "xen,uart_hvc";
 		status = "okay";
 	};
```
Space before new node

oleksiimoisieiev replied:
I still suggest to use overlays instead of changing dts. This is the way it was done in zephyr/samples/subsys/fs/littlefs. Why shouldn't we keep things this way?

firscity commented on boards/arm64/xenvm/xenvm-qemu.dts:
```diff
@@ -82,4 +83,45 @@
+	};
+};
+
+&flash_sim0 {
```
Do we actually need to edit previous node by handle in same dts?

firscity commented on boards/arm64/xenvm/xenvm.dts:
```diff
@@ -79,4 +80,45 @@
 		compatible = "xen,uart_hvc";
 		status = "okay";
 	};
```
Same comments as for xenvm-qemu

firscity commented on boards/arm64/xenvm/xenvm-qemu.dts:
```diff
@@ -82,4 +83,45 @@
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+/*
```
AFAIK, block comment should be also indented.

firscity commented on subsys/storage/flash_map/flash_map.c:
```diff
@@ -16,7 +16,6 @@
 #include <zephyr/storage/flash_map.h>
 #include "flash_map_priv.h"
 #include <zephyr/drivers/flash.h>
-#include <soc.h>
```
Do we actually need changes to Zephyr mainline kernel? If so, at least it should be added in separate commit with proper explanation, I don't understand the reason of such change. It may break other SOC.

dsemenets replied:
I don't know why this line present in those files. There are no dependency from soc.h.
Technically we can create empty soc.h for our board but why?

firscity replied:
I see only 2 valid approaches:
1. Create empty soc.h header to comply mainline code
2. Remove this includes from subsys with proper explanation and try to send it to mainline repo
Any of these changes should be in separate commit with explanation in commit message.

oleksiimoisieiev replied:
I believe this is an error of the author. Went through all of the soc.h files and see the following:
1) soc,h for different platforms have completely different definitions. Nothing common at all
2) Not all platforms have soc.h files which means this header is optional.
I think that common code is not a correct place for optional header without common interface.
I suggest to contact an author Andrzej Puzdrowski and ask him specifically. Who knows, maybe he will accept this change fast.

firscity commented on subsys/storage/flash_map/flash_map_layout.c:
```diff
@@ -15,7 +15,6 @@
 #include <zephyr/device.h>
 #include <zephyr/storage/flash_map.h>
 #include <zephyr/drivers/flash.h>
-#include <soc.h>
```
Same

firscity reviewed (changes requested):
Also, please rebase changes on top of aos-wip branch.

dsemenets committed:
subsys/storage/flash_map: remove unnecessary includes

Those files includes soc.h header which exists not for all boards.
soc.h consists soc-depended defenitions and need to be included by
soc-depended sources

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Reviewed-by: Dmytro Firsov <dmytro_firsov@epam.com>

firscity reviewed (approved):
`Reviewed-by: Dmytro Firsov <dmytro_firsov@epam.com>`

firscity commented:
@dsemenets, please re-push commit with added Reviewed-by and it will be merged.
